pkg/store/entities.go

```diff
@@ -2,54 +2,123 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-// Package store contains implementation of the store for persisting pipeline entities
 package store
 
 import (
-	"context"
-	"github.com/onosproject/onos-api/go/onos/topo"
 	p4api "github.com/p4lang/p4runtime/go/p4/v1"
 )
 
-// EntityStore is an abstraction of a store capable of maintaining various P4 entities
-type EntityStore interface {
-	// ID returns the ID of the device whose control entities it persists.
-	ID() topo.ID
+/*
+Notes:
 
-	// Read accepts a query in form of a list of partially populated entities and returns any
-	// matching entities on the specified channel
-	Read(ctx context.Context, query []*p4api.Entity, ch chan<- p4api.Entity) error
+Create a map per device, per:
+- each table (map[tableId]*table; map[hash(entry.matches)]*entry)
```

Comment:
Probably you already know this, in P4RT table entries are uniquely identified by the set of match fields **and** priority, so you probably need to add the priority to the hash